### DISCUSSION TITLE
Crazyradio usb driver installation on windows

### DISCUSSION THREAD

**Discussion by yuhsiangsu** [bitcraze/discussions#446] [Q&A]
Hi,

I am not able to instable the required USB driver for Crazyradio to run on my windows via Zadig. Is there any other way to install the USB driver? I have installed it successfully before, but somehow I accidentally delete the driver. And now I cannot install it back again. Please see below the error code I received from Zadig.

Thank you.

Zadig 2.7.765
Windows 10 Enterprise, 64-bit (Build 19042.1826)
ini file 'zadig.ini' not found in '\\nask.man.ac.uk\home$\Downloads' - default parameters will be used
default driver set to 'WinUSB'
0 devices found.
1 device found.
libwdi:debug [wdi_create_list] Hardware ID: USB\VID_1915&PID_7777&REV_0053
libwdi:debug [wdi_create_list] Compatible ID: USB\Class_FF&SubClass_FF&Prot_00
libwdi:debug [wdi_create_list] Driverless USB device (0): USB\VID_1915&PID_7777\09580F32B9
libwdi:debug [wdi_create_list] Device description: 'Crazyradio PA USB Dongle'
Using inf name: Crazyradio_PA_USB_Dongle.inf
Could not extract files
Driver Installation: FAILED (System policy has been modified to reject unsigned drivers)
libwdi:info [wdi_prepare_driver] K driver available - adding the libusbK DLLs to the libusb-win32 inf
libwdi:info [extract_binaries] Successfully extracted driver files to 'C:\Users\p37781ys\usb_driver'
libwdi:info [wdi_prepare_driver] Successfully created 'C:\Users\p37781ys\usb_driver\Crazyradio_PA_USB_Dongle.inf'
libwdi:info [wdi_prepare_driver] Creating and self-signing a .cat file...
libwdi:info [wdi_prepare_driver] Test signing is: Disabled
libwdi:debug [AddFileHash] 'libusb0.dll': PE type
libwdi:info [ScanDirAndHash] added hash for 'C:\Users\p37781ys\usb_driver\amd64\libusb0.dll'
libwdi:debug [AddFileHash] 'libusb0.sys': PE type
libwdi:info [ScanDirAndHash] added hash for 'C:\Users\p37781ys\usb_driver\amd64\libusb0.sys'
libwdi:debug [AddFileHash] 'libusb0_x86.dll': PE type
libwdi:info [ScanDirAndHash] added hash for 'C:\Users\p37781ys\usb_driver\amd64\libusb0_x86.dll'
libwdi:debug [AddFileHash] 'libusbk.dll': PE type
libwdi:info [ScanDirAndHash] added hash for 'C:\Users\p37781ys\usb_driver\amd64\libusbk.dll'
libwdi:debug [AddFileHash] 'libusbk_x86.dll': PE type
libwdi:info [ScanDirAndHash] added hash for 'C:\Users\p37781ys\usb_driver\amd64\libusbk_x86.dll'
libwdi:debug [AddFileHash] 'crazyradio_pa_usb_dongle.inf': INF type
libwdi:info [ScanDirAndHash] added hash for 'C:\Users\p37781ys\usb_driver\crazyradio_pa_usb_dongle.inf'
libwdi:debug [AddFileHash] 'libusb0.dll': PE type
libwdi:info [ScanDirAndHash] added hash for 'C:\Users\p37781ys\usb_driver\x86\libusb0.dll'
libwdi:debug [AddFileHash] 'libusb0.sys': PE type
libwdi:info [ScanDirAndHash] added hash for 'C:\Users\p37781ys\usb_driver\x86\libusb0.sys'
libwdi:debug [AddFileHash] 'libusb0_x86.dll': PE type
libwdi:info [ScanDirAndHash] added hash for 'C:\Users\p37781ys\usb_driver\x86\libusb0_x86.dll'
libwdi:debug [AddFileHash] 'libusbk.dll': PE type
libwdi:info [ScanDirAndHash] added hash for 'C:\Users\p37781ys\usb_driver\x86\libusbk.dll'
libwdi:info [CreateCat] Successfully created file 'C:\Users\p37781ys\usb_driver\Crazyradio_PA_USB_Dongle.cat'
libwdi:debug [CreateSelfSignedCert] Set Enhanced Key Usage, URL and CPS
libwdi:warning [CreateSelfSignedCert] Could not obtain a key container: Object already exists. (0x8009000F)
libwdi:error [wdi_prepare_driver] Could not sign cat file

**Comment by ataffanel** [comment]
When Windows introduces the requirement of signing all drivers, it was possible to install unsigned driver by restarting in safe mode and installing the drivers in safe mode. It seems to be possible in Windows 10 to [disable driver signature requirement](https://support.viewsonic.com/en/support/solutions/articles/33000252416-how-to-install-unsigned-drivers-in-windows-10). Zadig seems to start by trying to install the driver unsigned so if disabling driver signature check is an option for you, it should make zadig work.

I am not aware of other ways to get the libusb driver in Windows. In my search for the error I see that you have made a ticket on the libwdi repos, I would agree with the comment there, there seems to be something specific about your system blocking zadig.
